Libraries/Components/FocusableWindows/FocusableWindows.windows.js

```diff
@@ -0,0 +1,331 @@
+ *
+ *   const FocusableSomeComponent = createFocusableComponent(SomeComponent);
+ * 
+ *   class Whatever extends React.Component {
+ *     render () {
+ *         return (
+ *            <FocusableSomeComponent
+ *               {...this.props }
+ *               onKeyDown={ ... }
+ *               onFocus={ ... }
+ *               handledKeyDownKeys: {[FocusableSomeComponent.keys.Enter]}
+ *            >
+ *               ... .
+ *            </FocusableSomeComponent>
+ *         );
+ *      }
+ *   }
+ * 
+ * The generated component uses a helper FocusableWindow component that relies on WindowsControl native view.
+ * Implementation allows for splitting the property/styles (setNativeProps included )between WindowsControl and the original Component appropriately.
+ *
+ * @providesModule FocusableWindows
+ */
+'use strict';
+
+var PropTypes = require('prop-types');
+var React = require('React');
+var ReactNative = require('ReactNative');
+var ViewPropTypes = require('ViewPropTypes');
+var requireNativeComponent = require('requireNativeComponent');
+var UIManager = require('UIManager');
+const flattenStyle = require('flattenStyle');
+
+// This describes the propType based interface for WindowsControl
+class FocusableWindowsTemplate {
+  static focusablePropTypes = {
+    /**
+     * Controls whether the view is a tab stop. Useful for buttons and other
+     * controls that can be focused.
+     *
+     * @platform windows
+     */
+    isTabStop: PropTypes.bool,
+
+    /**
+     * Sets the tab index for the view.
+     *
+     * @platform windows
+     */
+    tabIndex: PropTypes.number,
+
+    /**
+     * Sets the keyboard navigation mode for the view.
+     *
+     * @platform windows
+     */
+    tabNavigation: PropTypes.oneOf([
+      'local',
+      'cycle',
+      'once'
+    ]),
+
+    /**
+     * Controls whether control should use system default provided focus rects
+     *
+     * @platform windows
+     */
+    disableSystemFocusVisuals: PropTypes.bool,
+
+    /**
+     * Called when the view receives focus.
+     *
+     * @platform windows
+     */
+    onFocus: PropTypes.func,
+
+    /**
+     * Called when the view focus is lost.
+     *
+     * @platform windows
+     */
+    onBlur: PropTypes.func,
+
+    /**
+     * Set of keys that should be handled on key down by this component.
+     *
+     * @platform windows
+     */
+    handledKeyDownKeys: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * Set of keys that should be handled on key up by this component.
+     *
+     * @platform windows
+     */
+    handledKeyUpKeys: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * Called when key down while component has focus.
+     *
+     * @platform windows
+     */
+    onKeyDown: PropTypes.func,
+
+    /**
+     * Called when key up while component has focus.
+     *
+     * @platform windows
+     */
+    onKeyUp: PropTypes.func,
+    /**
+     * A callback based "ref"-like property for the embedded Component
```

Comment:
>A callback based "ref"-like property for the embedded Component [](start = 7, length = 63)

Do you think it's worth calling out the gotcha with `setNativeProps`?

Reply:
Yes, will do, never hurts.

---
In reply to: [160785216](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r160785216) [](ancestors = 160785216)